getting a warning about a it saying something in async is deprecated

Just a little heads up I'm getting this warning in Visual Studio Code on Ubuntu 20.04 compiling with GCC 11
And I'm using Cmake to include the header only library as a INTERFACE

I have tried to clone a brand new up to date version of your repo and it still gives me the warning

Btw I do store all external libraries in the project folder in Libraries/External for easier management and that is why the path is starting with that

```
[build] Libraries/External/Cpp-socketwrapper/include/socketwrapper/detail/async.hpp: In member function ‘bool net::detail::async_context::add(int, net::detail::async_context::event_type, CALLBACK_TYPE&&)’:
[build] Libraries/External/Cpp-socketwrapper/include/socketwrapper/detail/async.hpp:125:38: warning: bitwise operation between different enumeration types ‘const net::detail::async_context::event_type’ and ‘EPOLL_EVENTS’ is deprecated [-Wdeprecated-enum-enum-conversion]
[build]   125 |             item.event.events = type | EPOLLET;
[build]       |                                      ^
```

And noticed you had made a issue about a month ago about async so were starting to think it were related wich is why I wanted to inform you

--- a/example/Makefile
+++ b/example/Makefile
@@ -1,7 +1,7 @@
-#cc = g++
+# cc = g++
 cc = clang++
 
-CFLAGS = -std=c++17 -fpic -Wall -Werror -Wextra -pedantic -g
+CFLAGS = -std=c++17 -fpic -Wall -Werror -Wextra -pedantic
 LDFLAGS = -lpthread
 
 %.o: %.cpp

--- a/include/socketwrapper/detail/async.hpp
+++ b/include/socketwrapper/detail/async.hpp
@@ -23,7 +23,7 @@ namespace detail {
 class async_context
 {
 public:
-    enum event_type
+    enum event_type : uint32_t
     {
         READ = EPOLLIN,
         WRITE = EPOLLOUT
@@ -122,7 +122,7 @@ public:
                 return false;
 
             auto& item = insert_it->second;
-            item.event.events = type | EPOLLET;
+            item.event.events = static_cast<uint32_t>(type) | EPOLLET;
             item.event.data.fd = sock_fd;
             item[type] = std::forward<CALLBACK_TYPE>(callback);
 

